test/unit.test.ts

```diff
@@ -1,20 +1,62 @@
+import * as path from 'path';
+import * as os from 'os';
 import chai = require('chai');
 import chaiAsPromised = require('chai-as-promised');
 import source = require('../');
+import resolvePath from './helpers/resolve-path';
+import * as semver from 'semver';
 
 const expect = chai.expect;
 chai.use(chaiAsPromised);
 
 describe('unit tests', () => {
   describe('__broccoliGetInfo__ API', () => {
+    it('normalizes relative paths', () => {
+      let node = new source.Directory('some/path', 'true', {
+        name: 'SomeName',
+        annotation: 'some annotation',
+      });
+
+      let pluginInterface = node.__broccoliGetInfo__();
+
+      expect(pluginInterface).to.have.property('sourceDirectory', resolvePath('some/path'));
+    });
+
+    it('passes through absolute + namespaced paths', () => {
+      let node = new source.Directory(resolvePath('some/path'), 'true', {
+        name: 'SomeName',
+        annotation: 'some annotation',
+      });
+
+      let pluginInterface = node.__broccoliGetInfo__();
+
+      expect(pluginInterface).to.have.property('sourceDirectory', resolvePath('some/path'));
+    });
+
+    it('passes namespaces absolute paths', () => {
+      let inputPath = path.resolve('some/path');
+      let node = new source.Directory(inputPath, 'true', {
+        name: 'SomeName',
+        annotation: 'some annotation',
+      });
+
+      let pluginInterface = node.__broccoliGetInfo__();
+
+      expect(pluginInterface).to.have.property('sourceDirectory', resolvePath('some/path'));
+
+      if (os.platform() === 'win32' && semver.gte(process.version, '9.0.0')) {
+        expect(pluginInterface.sourceDirectory).to.not.equal(inputPath);
```

Comment:
Could we add a comment or message for this case

Reply:
I removed the toNamespacedPath stuff for now (windows CI failed with Broccoli 0.16). I'll be bringing it back in a separate PR.